### PR TITLE
Coverity issues

### PR DESCRIPTION
**Pull request by ricab**
Addresses #2246.

### CODE REVIEW

File: Source/Urho3D/Resource/Image.cpp [hunk 1]
```diff
@@ -965,6 +965,7 @@ bool Image::LoadColorLUT(Deserializer& source)
     if (components != 3)
     {
         URHO3D_LOGERROR("Invalid image format, can not load image");
+        FreeImageData(pixelDataIn);
```
Comment (ricab) [inline Source/Urho3D/Resource/Image.cpp]: @weitjong, concerning this: while Coverity was down, I put a carefully coded [scope guard](https://ricab.github.io/scope_guard/) together. I would be happy to add to Urho3D. It is very useful in this sort of situation with early returns.

**Comment by ricab** [comment]
Hmm, travis build failed. It appears not to be using C++11: `error: no template named 'is_trivially_default_constructible' in namespace 'std'`

Is it a configuration issue?

File: Source/Tools/SpritePacker/SpritePacker.cpp [hunk 2]
```diff
@@ -328,7 +328,8 @@ void Run(Vector<String>& arguments)
     spriteSheetImage.SetSize(packedWidth, packedHeight, 4);
 
     // zero out image
-    spriteSheetImage.SetData((unsigned char*)calloc(sizeof(unsigned char), (size_t)packedWidth * packedHeight * 4));
+    // Ugly: depends on implementation details - is this really needed?
+    memset(spriteSheetImage.GetData(), 0, packedWidth * packedHeight * 4);
```
Comment (weitjong) [inline Source/Tools/SpritePacker/SpritePacker.cpp]: Incidentally I have just fixed this leak yesterday in the `clang-tidy-up` dev branch. So, you could remove this fix in your PR.

File: Source/Urho3D/Resource/Image.cpp (hunk 1 above)
Comment (weitjong) [reply Source/Urho3D/Resource/Image.cpp]: Sounds good to me.

**Comment by weitjong** [comment]
All our CI support C++11 standard now. The failure you saw is really our linter starts to do its work. Scroll down the build log to see the root cause.

File: Source/Urho3D/Resource/Image.cpp (hunk 1 above)
Comment (ricab) [reply Source/Urho3D/Resource/Image.cpp]: alright, I opened a separate PR

**Comment by ricab** [comment]
Closing: this is now superseded by #2281

**Pull request closed without merging**